### PR TITLE
[auth][test] Add unit test basis for auth service

### PR DESCRIPTION
**Pull request by jaylees14**
Wow such verbosity.

Add all the boilerplate we need to write tests for the auth service.

### CODE REVIEW

File: auth/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	if !ok {
+		t.Fatalf("Token doesn't contain an access token: %s", err.Error())
+	}
+
+	token, _, err := new(jwt.Parser).ParseUnverified(rawToken, jwt.MapClaims{})
+	if err != nil {
+		t.Fatalf("Could not decode JWT: %s", err.Error())
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
```
Comment (c-harding) [inline auth/auth_test.go:103]: What is this syntax?

**Review by c-harding — APPROVED** [review]

File: auth/auth_test.go (hunk 1 above)
Comment (jaylees14) [reply auth/auth_test.go:103]: https://golang.org/ref/spec#Type_assertions

**Review by Lewiky — APPROVED** [review]

File: auth/go.mod [hunk 2]
```diff
@@ -4,8 +4,8 @@ go 1.13
 
 require (
 	github.com/asaskevich/govalidator v0.0.0-20200108200545-475eaeb16496
-	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible
```
Comment (willjvsmith) [inline auth/go.mod:7]: What's the `incompatible` referring to?

**Review by willjvsmith — APPROVED** [review]

File: auth/go.mod (hunk 2 above)
Comment (jaylees14) [reply auth/go.mod:7]: https://github.com/dgrijalva/jwt-go/issues/294 Looks like it's a `go mod` thing, where `jwt-go` doesn't fully support modules in v3, but will do in v4. Looks like it's in preview for now, but we can upgrade if/when they release, although they mention will be some breaking changes

**Pull request merged**